src/version/common/Topbar/fragments/UserPanelButton.js

```diff
@@ -0,0 +1,105 @@
+import React, { Component } from "react";
+import { get } from "lodash";
+import { connect } from 'react-redux';
+
+import { withStyles } from "@material-ui/core/styles";
+import Typography from "@material-ui/core/Typography";
+import IconButton from '@material-ui/core/IconButton';
+import Popover from '@material-ui/core/Popover';
+import Card from '@material-ui/core/Card';
+import PersonIcon from '@material-ui/icons/Person';
+
+import { userInfoAction } from "../../../../core/actions/userInfoAction";
+import CustomLogoutButton from "../../../../core/common/Buttons/CustomLogoutButton";
+
+const styles = {
+    userPanel: {
+        minWidth: 220,
+        padding: 12,
+    },
+    userName: {
+        marginBottom: 7,
+        fontSize: 18,
+        fontWeight: 800,
+    },
+    userRole: {
+        fontSize: 14,
+        marginBottom: 7,
+    }
+};
+
+/**
+ * This component returns User panel popover
+ *
+ * @author Bogdan Shcherban <bsc@piogroup.net>
+ */
+class UserPanelButton extends Component {
+
+    state = {
+        anchorEl: null,
+    };
+
+    handleMenu = event => {
+        this.setState({ anchorEl: event.currentTarget });
+    };
+
+    handleClose = () => {
+        this.setState({ anchorEl: null });
+    };
+
+    componentDidMount() {
+        this.props.getUserInfo();
+    }
+
+    render() {
+        const { classes, logout, userInfo } = this.props;
+        const { anchorEl } = this.state;
+        const isTopbarMenuOpen = Boolean(anchorEl);
```

Comment:
not fan of it. You'd better store `isOpen` in the state and use React refs for handling the anchor element.

Reply:
Done via refs